src/Events/Custom_Tables/V1/WP_Query/Provider.php

```diff
@@ -48,15 +51,41 @@ public function register() {
 			add_action( 'tribe_repository_events_init', [ $this, 'replace_repository_query_filters' ] );
 		}
 
+		//
+		if ( ! has_filter( 'tec_events_custom_tables_v1_query_modifier_implementations', [
```

Comment:
See my comment in the trait and classes using it: this adds complexity that is not required. The `WP_Query_Monitor` and `Custom_Tables_Query` monitor in ECP are the base, and it's fine for them to define the base implementation set.